p2p/logger.go

```diff
@@ -0,0 +1,72 @@
+package p2p
+
+import "go.uber.org/zap"
+
+// SugaredLoggerInterface a interface for zap sugared logger,
+// use this can allow some project based on eos-go use it's different logger imp
+type SugaredLoggerInterface interface {
+
```

Comment:
Why import the interface if we're already dependent on zap ?

Reply:
at first i wannot to change the log format in p2p package, just hide, so a interface can allow someone to use diff logger, like this : [https://github.com/uber-go/zap/blob/master/FAQ.md#why-arent-logger-and-sugaredlogger-interfaces](url), but if we use structured logging by zap, i think it is no need to add a interface.

Reply:
It's such a huge Interface that I think it's not that useful in the end. I'd prefer having only an io.Writer as what is passed in by users of this lib. The zap.Logger can then use that as its output.